network/transport/v2/protocol.proto

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2021. Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+syntax = "proto3";
+
+option go_package = "github.com/nuts-foundation/nuts-node/network/transport/v2";
+
+package v2;
+
+service Protocol {
+  rpc Stream (stream Message) returns (stream Message) {
```

Comment:
Currently, we're using the method name (in this case `Stream`) in the `grpcOutboundStream`/`grpcInboundStream` maps right? Wouldn't it make more sense to call this something like: `ConnectV2` then?

Reply:
I want to stay clear of `Connect` since we have connections and connection managers all over the place. Maybe `Invoke`, `Exchange`?

Reply:
Both would be fine but it should have the `V2` suffix right? Otherwise the v3 protocol has to choose a different method name. That would be a mess. Or we should choose a different key for the `grpcOutboundStream` and inbound stream maps.

Reply:
> Both would be fine but it should have the V2 suffix right? Otherwise the v3 protocol has to choose a different method name. That would be a mess. Or we should choose a different key for the grpcOutboundStream and inbound stream maps.

The package name is already `v2`

Reply:
> > Both would be fine but it should have the V2 suffix right? Otherwise the v3 protocol has to choose a different method name. That would be a mess. Or we should choose a different key for the grpcOutboundStream and inbound stream maps.
> 
> The package name is already `v2`

I'm aware. That's not what I mean. If you look at this for example: https://github.com/nuts-foundation/nuts-node/pull/601/files#diff-56b8a578980a5f6863c2b3e7dc55ca89c600a67014b54b4287fc7e4dfded819cR154

You'll see that the gRPC method name is used to differentiate different gRPC streams. This means that if a v2 protocol gRPC stream will be opened it will be stored in `grpcOutboundStreams["Stream"]`. For v1 this will be: `grpcOutboundStreams["Connect"]`. This doesn't really make sense to me as it says nothing about the protocol version and assumes new protocol versions always have a different method (which is not in any way a requirement).

Reply:
Less generic could be very helpful when doing a global find on the exact term. `stream` would get to many hits.
`InitiateV2ProtocolChannel` (but then in German because they are master in combining words.)

Reply:
The `method` format is `/<protobuf package>.<service>/<call>`, so for v1:

`transport.Network/Connect`

and for v2:

`/v2.Protocol/Stream`

(I've got a feeling that it's also the HTTP/2 URL which is used...)

They need to be unique, so I think it's good enough?

Reply:
Check. Wasn't aware that the service name was already present in the method name